integration-tests/google-pubsub/src/test/java/org/apache/camel/quarkus/component/google/pubsub/it/GooglePubsubTest.java

```diff
@@ -16,18 +16,34 @@
  */
 package org.apache.camel.quarkus.component.google.pubsub.it;
 
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
 import io.quarkus.test.common.QuarkusTestResource;
 import io.quarkus.test.junit.QuarkusTest;
 import io.restassured.RestAssured;
+import org.apache.camel.quarkus.test.support.google.GoogleCloudTestResource;
+import org.hamcrest.Matchers;
+import org.jboss.logging.Logger;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.api.condition.DisabledIfEnvironmentVariable;
 
-import static org.hamcrest.CoreMatchers.is;
+import static org.awaitility.Awaitility.await;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
 @QuarkusTest
-@QuarkusTestResource(GooglePubsubTestResource.class)
+@QuarkusTestResource(GoogleCloudTestResource.class)
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
 class GooglePubsubTest {
+    private static final Logger LOG = Logger.getLogger(GooglePubsubTest.class);
 
     @Test
+    @Order(1)
```

Comment:
Is it needed to order tests ? What happen if they are executed in the stated order ?

Reply:
Ordering of the test was one of my attempts to find a reason of CI failures (in case that they are caused by different ordering). Unfortunately ordering didn't help, so I will remove it.

Reply:
I'm keeping the ordering of the tests, because it is a workaround of the https://github.com/apache/camel-quarkus/issues/3957. I mentioned it also as the comment in the test class.